packages/tinacms/src/toolkit/fields/components/reference/reference-select.tsx

```diff
@@ -95,55 +115,93 @@ const useGetOptionSets = (cms: TinaCMS, collections: string[]) => {
+    input.onChange(value)
+  }, [value, input, optionSets])
 
   if (loading === true) {
     return <LoadingDots color="var(--tina-color-primary)" />
   }
 
   return (
     <>
-      <select
-        ref={ref}
-        id={input.name}
-        value={input.value}
-        onChange={input.onChange}
-        className={selectFieldClasses}
-        {...input}
-      >
-        <option value={''}>Choose an option</option>
-        {optionSets.length > 0 &&
-          optionSets.map(({ collection, edges }: OptionSet) => (
-            <optgroup key={`${collection}-group`} label={collection}>
-              {edges.map(
-                ({
-                  node: {
-                    id,
-                    _internalSys: { title },
-                  },
-                }) => (
-                  <option key={`${id}-option`} value={id}>
-                    {title || id}
-                  </option>
-                )
-              )}
-            </optgroup>
-          ))}
-      </select>
-      <MdKeyboardArrowDown className="absolute top-1/2 right-3 w-6 h-auto -translate-y-1/2 text-gray-300 group-hover:text-blue-500 transition duration-150 ease-out" />
+      <Popover open={open} onOpenChange={setOpen}>
+        <PopoverTrigger asChild>
+          <Button
+            variant="outline"
+            role="combobox"
+            aria-expanded={open}
+            className="w-52 justify-between"
+          >
+            <p className="truncate">{displayText ?? 'Choose an option...'}</p>
+          </Button>
+        </PopoverTrigger>
+        <PopoverContent className="p-0 relative">
+          <Command>
+            <CommandInput placeholder="Search reference..." />
+            <CommandEmpty>No reference found</CommandEmpty>
+            {optionSets.length > 0 &&
+              optionSets.map(({ collection, edges }: OptionSet) => (
+                <CommandGroup key={`${collection}-group`} heading={collection}>
+                  <CommandList>
+                    {edges.map(
+                      ({
+                        node: {
+                          id,
+                          _internalSys: { filename },
+                          _values: { title, name },
+                        },
+                      }) => (
+                        <CommandItem
+                          key={`${id}-option`}
+                          value={id}
+                          onSelect={() => {
+                            setValue(id)
+                            setOpen(false)
+                          }}
+                        >
+                          <div className="flex flex-col">
+                            <span className="font-semibold text-sm">
+                              {title || name || id}
+                            </span>
+                            {(title || name) && (
+                              <span className="text-x">{filename}</span>
+                            )}
+                          </div>
```

Comment:
I think this is okay for now - but I think we can't 100% be sure they will have `name` or `title` lets improve this to add an option that the user can define that sets the `label` and `option component` So the users can customise this completely!